---
uint32 goalReached
uint32 goalCanceled
